frame/evm/src/lib.rs

```diff
@@ -679,6 +731,15 @@ impl<T: Config> GasWeightMapping for FixedGasWeightMapping<T> {
 					.base_extrinsic,
 			);
 		}
+
+		// Apply a gas to proof size ratio based on BlockGasLimit
+		let block_gas_limit: u64 = T::BlockGasLimit::get().unique_saturated_into();
+		let gas_ratio: sp_arithmetic::Perquintill =
```

Comment:
It's very inefficient to recompute the ratio at each execution, please make it an hardcoded constant.
To check the constant value, we can add a unit rust test that recompute the ratio or recompute it in a debug assertion:

`debug_assert_eq!(CONSTANT, fn_compute(params))`

Reply:
I don't understand, how can the gas ratio be a constant if the gas is user provided?

Reply:
The ratio BlockGasLimit / MaxPovSize is a constant

Reply:
done